srtcore/sync_cxx11.cpp

```diff
@@ -58,6 +58,11 @@ srt::sync::Condition::~Condition() {}
 
 void srt::sync::Condition::init() {}
 
+void srt::sync::Condition::reset()
+{
+    memset((void *) &m_cv, 0, sizeof(m_cv));
```

Comment:
Whatever you'd do here is tricky, but there's one thing that you can do to avoid any risks:

1. Create a local variable of the same type (in this case it's `std::condition_variable`).
2. Copy the contents of the freshly created variable into `m_cv`.

By doing this you at least avoid a risk of creating some contents that the system wouldn't ever do as initial object state and go crazy when trying to destroy it.